src/dagmc/dagmcmetadata.cpp

```diff
@@ -0,0 +1,591 @@
+#include "dagmcmetadata.hpp"
+#include <iostream>
+#include <set>
+#include <algorithm>
+
+// constructor for metadata class
+dagmcMetaData::dagmcMetaData(moab::DagMC* dag_ptr, bool verbosity) {
+  DAG = dag_ptr; // dagmc pointer
+  verbose = verbosity;
+  // these are the keywords that dagmc will understand
+  // from groups if you need to process more
+  // they should be added here
+  metadata_keywords.push_back("mat");
+  metadata_keywords.push_back("rho");
+  metadata_keywords.push_back("boundary");
+  metadata_keywords.push_back("tally");
+  metadata_keywords.push_back("importance");
+
+  // allow some synonyms
+  keyword_synonyms[ "rho" ] = "density";
+  keyword_synonyms[ "mat" ] = "material";
+}
+
+// destructor doesnt need to do anything
+dagmcMetaData::~dagmcMetaData() {
+}
+
+// load the property data from the dagmc instance
+void dagmcMetaData::load_property_data() {
+  parse_material_data();
+  parse_importance_data();
+  parse_boundary_data();
+  parse_tally_volume_data();
+  parse_tally_surface_data();
+  // finalise_counters();
```

Comment:
Recommend removing commented out code, or leaving a note indicating under what conditions it can be removed.